source/developer/fx-guidelines.rst

```diff
@@ -242,134 +351,113 @@ Input: Hints
 
-    ..  image:: ../images/inputField2.png
-        :alt: Radio Buttons
+**Example 3:**
+    And other textareas or inputs should also perform their primary action when "Enter" is pressed, here's an example of the "Edit Header" modal.
+
+    ..  image:: ../images/inputBehaviour3.png
+        :alt: Edit header modal
         :width: 500 px
 
 -----
 
-Display: Button Placement
--------------------------
-
-**Dialog BOTTOM RIGHT BUTTONS should be either “Close”, or “Cancel” and [ACTION_BUTTON].**
-
-If there’s one button on the bottom right, it should be “Close”, if there are two, the one on the left should be “Cancel” and the one on the right should be an [ACTION_BUTTON], like “Save” or “Send Invitations”.
+Input: Validation
+-----------------
 
-**Example:**
+**Feedback on action:**
 
-    Correct:
-        Single button at the bottom right should be “Close”.
+The user should be notified about the action he performed along with any implications associated with it.
 
-        ..  image:: ../images/buttonPlacement1.png
-            :alt: Button Placement 1
-            :width: 500 px
+  Example 1:
+    "Settings are saved but will be applied after a server restart"
 
-    Correct:
-        When there are two buttons on bottom right, left button should be “Cancel” and the button on the right should be the [ACTION_BUTTON], in this case “Send Invitations”.
+      ..  image:: ../images/confirm2.jpg
+          :alt: Settings saved
 
-        ..  image:: ../images/buttonPlacement2.png
-            :alt: Button Placement 2
-            :width: 500 px
+  Example 2:
+    "Link copied to clipboard"
 
-    Incorrect:
-        When there are two buttons at the bottom right, left button should not be “Close”, as it’s not clear if closing will or won’t execute the [ACTION_BUTTON].
+      ..  image:: ../images/confirm1.jpg
+          :alt: Confirmation message
 
-        ..  image:: ../images/buttonPlacement3.png
-            :alt: Button Placement 3
-            :width: 500 px
+**Save prompt:**
 
------
+A prompt should appears if a user makes changes to a setting and attempts to navigate away without saving them.
 
-Display: Number of Choices
---------------------------
+  Example:
+    "You have unsaved changes, are you sure you want to discard them?"
 
-To simplify decisions, when practical, limit the number of choices to 3 or 4, and add separators or headings between logical groups. See `Hick’s Law <https://en.wikipedia.org/wiki/Hick%27s_law>`_ for background on why this helps.
+      ..  image:: ../images/save1.png
+          :alt: Save prompt
 
-Example:
+---------------------------
 
-    Incorrect:
-        No clear separation between distinct options.
+Menu items
+----------
 
-        ..  image:: ../images/choices1.png
-            :alt: No separation
-            :width: 500 px
+Items in a menu should be grouped together based on the category they belong to. Items per category however should not exceed 5 for the app and 7 for system console.
```

Comment:
Propose: "...should not exceed five, with the execption of seven for the system console."